routing/router.go

```diff
@@ -1523,6 +1523,10 @@ type LightningPayment struct {
 	// if there isn't a route with lower fees than this limit.
 	FeeLimit lnwire.MilliSatoshi
 
+	// CltvLimit is the maximum time lock that is allowed for attempts to
+	// complete this payment.
+	CltvLimit *uint32
```

Comment:
I encountered this pattern for the `outgoingChannelID *uint64` earlier. Could we instead let a zero-value indicating it is unset? It really makes serialization easier :)

(I changed this for `outgoingChannelID`  in my `router reliable payments` branch, as I need to serialize the payment attempts.)

Reply:
I am generally wary of magic values like 0. I use the pointer type be explicit about whether the option applies or not. I prevents people from forgetting to handle the 'not set' case. A channel id of 0 and a cltv limit of 0 are values that are valid. It is just that they will probably never be used in practice.

Reply:
For example with the fee limit it became confusing. A fee limit of zero means no limit. But a zero fee limit could also mean that we may only pay to a direct connection (fee zero).

Reply:
Yep, it's a tradeoff. I definitely see the advantages of a value explicitly _not_ being set. I'm okay with this approach for now, as the serialization I mentioned earlier has been deferred.